lib/storage/file.js

```diff
@@ -65,14 +64,17 @@ var STORAGE_UPLOAD_BASE_URL = 'https://www.googleapis.com/upload/storage/v1/b';
  * @alias module:storage/file
  * @constructor
  */
-function File(bucket, name, metadata) {
+function File(bucket, name, options) {
   if (!name) {
     throw Error('A file name must be specified.');
   }
 
+  options = options || {};
+
   this.bucket = bucket;
+  this.explicitGeneration = parseInt(options.generation, 10);
```

Comment:
I don't see an issue of exposing `file.generation` or even just `file.metadata.generation` but I wouldn't expose this as a value that someone can rely on (not that we would recommend it anyway)